uxarray/io/_ugrid.py

```diff
@@ -93,30 +93,42 @@ def _encode_ugrid(ds):
     """Encodes an unstructured grid represented under a ``Grid`` object as a
     ``xr.Dataset`` with an updated grid topology variable."""
 
-    if "grid_topology" in ds:
-        ds = ds.drop_vars(["grid_topology"])
+    # Create a copy of the input dataset to avoid modifying the original Grid._ds
+    out_ds = ds.copy()
```

Comment:
Good catch, but no need for use of `.copy()` and duplicate variables (i.e. `out_ds`) if you can keep the original code and replace only:

`ds["grid_topology"] = grid_topology_da` 

on Line 117 with the following: 

`ds = ds.assign(grid_topology=grid_topology_da)`

This way, since there is an assignment back to `ds`, it will not change the original `_ds` (will not behave like pass-by-reference)